### PR TITLE
Fix #21434: Number of guests overflows in objective text

### LINKED ISSUE #21434: Guest Objective Overflows when reaching past 32,750
### Operating System

Windows 10, Windows 11, 64-Bit

### OpenRCT2 build

OpenRCT2

### Base game

RollerCoaster Tycoon 2

### Area(s) with this issue?

This is a development issue

### Describe the issue

When the guest objective goes past 32,750, it overflows and rolls back to -32,736 as any number past 32,768
 (the signed 16-bit integer limit)Overflow

### Steps to reproduce

1. Open Scenario Editor
2. Make the objective certain guests in the park
3. Put the input above 32,768 guests

### Attachments

_No response_

### PR DESCRIPTION
**Pull request by jan-kelemen**
Format number of guests in objective as Comma32 instead of Comma16.

Fixes [#21434]

### CODE REVIEW

File: src/openrct2-ui/windows/Park.cpp [hunk 1]
```diff
@@ -1099,7 +1099,12 @@ class ParkWindow final : public Window
         }
         else
         {
-            ft.Add<uint16_t>(gameState.ScenarioObjective.NumGuests);
+            if (gameState.ScenarioObjective.Type == OBJECTIVE_GUESTS_BY
+                || gameState.ScenarioObjective.Type == OBJECTIVE_GUESTS_AND_RATING)
+                ft.Add<int32_t>(gameState.ScenarioObjective.NumGuests);
+            else
+                ft.Add<uint16_t>(gameState.ScenarioObjective.NumGuests);
```
Comment (duncanspumpkin) [inline src/openrct2-ui/windows/Park.cpp:1106]: What? are we using this field for multiple different things?

Comment (duncanspumpkin) [reply src/openrct2-ui/windows/Park.cpp:1106]: Ahh i see what is happening. I don't like this there is a high chance that it will go wrong.

Comment (jan-kelemen) [reply src/openrct2-ui/windows/Park.cpp:1106]: As far as I can tell from the usages no, but this code is used to print multiple messages, i.e the next string which uses the same code pops 16 bytes off,
```
STR_2392    :{BLACK}To achieve a monthly income from ride tickets of at least {POP16}{POP16}{CURRENCY}
```

Comment (duncanspumpkin) [reply src/openrct2-ui/windows/Park.cpp:1106]: You see how OBJECTIVE_BUILD_THE_BEST is separated out just do the same for the objectives with guest counts

File: src/openrct2-ui/windows/Park.cpp [hunk 2]
```diff
@@ -1097,9 +1097,23 @@ class ParkWindow final : public Window
             }
             ft.Add<StringId>(rideTypeString);
         }
+        else if (
+            gameState.ScenarioObjective.Type == OBJECTIVE_GUESTS_BY
+            || gameState.ScenarioObjective.Type == OBJECTIVE_GUESTS_AND_RATING)
+        {
+            ft.Add<int32_t>(gameState.ScenarioObjective.NumGuests);
+            if (gameState.ScenarioObjective.Type == OBJECTIVE_GUESTS_BY)
```
Comment (duncanspumpkin) [inline src/openrct2-ui/windows/Park.cpp:1105]: at this point you may as well just do this at the higher level else if chain.

**Comment by jan-kelemen** [comment]
Resolved review comment, rebased to develop to resolve conflicts.

**Review by duncanspumpkin — APPROVED** [review]
Code looks fine haven't tested it though

**Review by Gymnasiast — APPROVED** [review]
Tested it, works correctly. Could you add a changelog entry?

**Comment by jan-kelemen** [comment]
Changelog entry added.

File: distribution/changelog.txt [hunk 3]
```diff
@@ -4,6 +4,7 @@
 - Change: [#21529] Classify “Southern Sands”, “Tiny Towers”, “Nevermore Park”, “Pacifica” as expert scenarios.
 - Fix: [#910] Extra viewport does not preserve the location when rotating.
 - Fix: [#21543] Crash with creating a TrackIterator with invalid arguments.
+- Fix: [#21434] Number of guests overflows in objective text.
```
Comment (Gymnasiast) [inline distribution/changelog.txt:7]: This should be one line up (the changelog is ordered by category and then by ID).

Comment (jan-kelemen) [reply distribution/changelog.txt:6]: Fixed.

**Review by Gymnasiast — APPROVED** [review]

**Comment by Gymnasiast** [comment]
@jan-kelemen Could you also create a PR against the Localisation repository that replaces the formatting token in all other languages?

**Comment by jan-kelemen** [comment]
@Gymnasiast yup, done https://github.com/OpenRCT2/Localisation/pull/2787

**Pull request merged**